app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala

```diff
@@ -272,9 +293,9 @@ user.incomeSources.getLatencyDetails(incomeSourceType, id.value) match {
     (backCall, successCall)
   }
 
-  private def getPostAction(taxYear: String, changeTo: String, isAgent: Boolean, incomeSourceType: IncomeSourceType): Call = {
+  private def getPostAction(taxYear: String, changeTo: String, incomeSourceType: IncomeSourceType): Call = {
     routes.ConfirmReportingMethodSharedController
-      .submit(taxYear, changeTo, isAgent, incomeSourceType)
+      .submit(taxYear, changeTo, incomeSourceType)
```

Comment:
This is the cause of your error. You need the submitAgent when the user is an agent.